numpy/random/mtrand/mtrand.pyx

```diff
@@ -1252,17 +1737,48 @@ cdef class RandomState:
         key = np.dtype(dtype).name
         if not key in _randint_type:
             raise TypeError('Unsupported dtype "%s" for randint' % key)
-        lowbnd, highbnd, randfunc = _randint_type[key]
+        lowbnd, highbnd, randfunc_sc, randfunc = _randint_type[key]
 
-        if low < lowbnd:
-            raise ValueError("low is out of bounds for %s" % (key,))
-        if high > highbnd:
-            raise ValueError("high is out of bounds for %s" % (key,))
-        if low >= high:
-            raise ValueError("low >= high")
+        try:
+            # Determine if `low` and `high` are both integers
+            # Otherwise, at least one of them is an array
+            test = int(low)
+            test = int(high)
+
+            if low < lowbnd:
+                raise ValueError("low is out of bounds for %s" % (key,))
+            if high > highbnd:
+                raise ValueError("high is out of bounds for %s" % (key,))
+            if low >= high:
+                raise ValueError("low >= high")
 
-        with self.lock:
-            return randfunc(low, high - 1, size, self.state_address)
+            with self.lock:
+                return randfunc_sc(low, high - 1, size, self.state_address)
+
+        except TypeError:
+            try:
+                test = int(low)
+                low_array = np.array([low])
+            except TypeError:
+                low_array = np.array(low)
+
+            try:
+                test = int(high)
+                high_array = np.array([high])
+            except TypeError:
+                high_array = np.array(high)
```

Comment:
I think these should just be `high_array = np.array(high)` and similarly for `low` -- wrapping integers into a list before passing to `array` is neither necessary nor desireable. (I guess in this case it doesn't actually cause any harm, because we already special-cased the scalar/scalar case above, so we know that on this path at least one of the inputs has >1 dimension and an array with shape `(1,)` will broadcast against any array with >1 dimension. But if we just call `high = np.array(high)`, then not only is the code simpler but this branch becomes correct in general, even if unprotected by the scalar checks above.)